modules/sw360/src/main/java/org/eclipse/sw360/antenna/sw360/utils/SW360ComponentAdapterUtils.java

```diff
@@ -71,6 +71,8 @@ public static void setName(SW360Component component, Artifact artifact) {
         final String name = createComponentName(artifact);
         if (!name.isEmpty()) {
             component.setName(name);
+        } else {
+            component.setName(artifact.toString()); // TODO: ugly hack
```

Comment:
Is there nothing better at all, like the filename, if there is one, or part of the coordinates? This creates a pretty horrible String

Reply:
> This creates a pretty horrible String

Yes, but it is the last fallback I can think of.

This should just the fallback if `createComponentName(..)` fails. One should improve that function to better support npm artifacts.

Reply:
if you create an issue for this as a follow up, i'd be okay with this :)